UpgradedVehicles/Patchers.cs

```diff
@@ -22,53 +27,86 @@ internal static void Postfix(ref float __result, GameObject target)
         }
     }
 
-    [HarmonyPatch(typeof(SeaMoth))]
-    [HarmonyPatch(nameof(SeaMoth.Awake))]
-    internal class SeaMoth_Awake_Patcher
+    [HarmonyPatch]
+    internal class VehiclePatches
     {
+#if SUBNAUTICA
+        [HarmonyPatch(typeof(SeaMoth))]
+        [HarmonyPatch(nameof(SeaMoth.Awake))]
         [HarmonyPrefix]
-        internal static void Prefix(ref SeaMoth __instance)
+        internal static void PreSeamothAwake(ref SeaMoth __instance)
         {
-            QuickLogger.Debug(nameof(SeaMoth_Awake_Patcher));
+            QuickLogger.Debug(nameof(PreSeamothAwake));
             VehicleUpgrader vehicleUpgrader = __instance.gameObject.EnsureComponent<VehicleUpgrader>();
             vehicleUpgrader.Initialize(ref __instance);
         }
-    }
 
-    [HarmonyPatch(typeof(Exosuit))]
-    [HarmonyPatch(nameof(Exosuit.Awake))]
-    internal class Exosuit_Awake_Patcher
-    {
+#elif BELOWZERO
+        [HarmonyPatch(typeof(SeaTruckMotor))]
+        [HarmonyPatch(nameof(SeaTruckMotor.Start))]
+        [HarmonyPostfix]
+        internal static void PostSeatruckStart(ref SeaTruckMotor __instance)
+        {
+            QuickLogger.Debug(nameof(PostSeatruckStart));
+            VehicleUpgrader vehicleUpgrader = __instance.gameObject.EnsureComponent<VehicleUpgrader>();
+            vehicleUpgrader.Initialize(ref __instance);
+        }
+#endif
+
+        [HarmonyPatch(typeof(Exosuit))]
+        [HarmonyPatch(nameof(Exosuit.Awake))]
         [HarmonyPrefix]
-        internal static void Prefix(ref Exosuit __instance)
+        internal static void PreExosuitAwake(ref Exosuit __instance)
         {
-            QuickLogger.Debug(nameof(Exosuit_Awake_Patcher));
+            QuickLogger.Debug(nameof(PreExosuitAwake));
             VehicleUpgrader vehicleUpgrader = __instance.gameObject.EnsureComponent<VehicleUpgrader>();
             vehicleUpgrader.Initialize(ref __instance);
         }
-    }
 
-    [HarmonyPatch(typeof(SeaMoth))]
-    [HarmonyPatch(nameof(SeaMoth.OnUpgradeModuleChange))]
-    internal class SeaMoth_OnUpgradeModuleChange_Patcher
-    {
+#if SUBNAUTICA
```

Comment:
Maybe not if Subnautica just incase someone ever gets the seamoth back in working order in BZ?